src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -432,4 +458,26 @@ private function loadFromExtensions($content)
             $this->container->loadFromExtension($namespace, $values);
         }
     }
+
+    /**
+     * Checks the keywords used to define a service.
+     *
+     * @param string $id         The service name
+     * @param array  $definition The service definition to check
+     *
+     * @throws InvalidArgumentException When an invalid keyword is used
+     */
+    private static function checkDefinition($id, array $definition)
+    {
+        foreach ($definition as $key => $value) {
+            if (!isset(static::$keywords[$key])) {
+                throw new InvalidArgumentException(sprintf(
+                    'The "%s" key in "%s" service definition is not valid. It must be one of "%s".',
+                    (string) $key,
```

Comment:
no need to cast here, and usually, we prefer putting exceptions on one single (long) line